controllers/external/util.go

```diff
@@ -81,11 +81,55 @@ func CloneTemplate(ctx context.Context, in *CloneTemplateInput) (*corev1.ObjectR
 	if err != nil {
 		return nil, err
 	}
-	template, found, err := unstructured.NestedMap(from.Object, "spec", "template")
+	generateTemplateInput := GenerateTemplateInput{
+		Template:    from,
+		Namespace:   in.Namespace,
+		ClusterName: in.ClusterName,
+		OwnerRef:    in.OwnerRef,
+		Labels:      in.Labels,
+	}
+	to, err := GenerateTemplate(generateTemplateInput)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the external clone.
+	if err := in.Client.Create(context.Background(), to); err != nil {
+		return nil, err
+	}
+
+	return GetObjectReference(to), nil
+}
+
+// GenerateTemplate input is everything needed to generate a new template.
+type GenerateTemplateInput struct {
+	// Template is the TemplateRef turned into an unstructured.
+	// +required
+	Template *unstructured.Unstructured
+
+	// Namespace is the Kuberentes namespace the cloned object should be created into.
+	// +required
+	Namespace string
+
+	// ClusterName is the cluster this object is linked to.
+	// +required
+	ClusterName string
+
+	// OwnerRef is an optional OwnerReference to attach to the cloned object.
+	// +optional
+	OwnerRef *metav1.OwnerReference
+
+	// Labels is an optional map of labels to be added to the object.
+	// +optional
+	Labels map[string]string
+}
+
+func GenerateTemplate(in GenerateTemplateInput) (*unstructured.Unstructured, error) {
```

Comment:
```suggestion
func GenerateTemplate(in *GenerateTemplateInput) (*unstructured.Unstructured, error) {
```

for consistency with the other function

Reply:
fixed!